### PR TITLE
Add support ssh:// URL and github local branch name fix.

### PR DESCRIPTION
**Pull request by kounoike**
by [Git manual](https://git-scm.com/docs/git-clone#_git_urls_a_id_urls_a) git command also supports ssh:// style URL.  It is useful for GitBucket supoprt(#144).

And Gogs/Gitea can use any user for ssh access. This change accepts any user on scp-style `user@server:pathto/repo.git`.

### CODE REVIEW

**Comment by guyzmo** [comment]
hum it's missing the third commit 😖

**Comment by kounoike** [comment]
> hum it's missing the third commit

"third commit" means local branch name fix? I will keep it to #144. (I think it has a little relation to GitBucket support).

**Comment by guyzmo** [comment]
merged 👍

File: git_repo/repo.py [hunk 1]
```diff
@@ -163,15 +163,15 @@ def _guess_repo_slug(self, repository, service, resolve_targets=None):
...
+                        url = url[:-4]
+                    if url.find('://') > -1:
+                        # http://, https:// and ssh://
                         *_, user, name = url.split('/')
                         self.set_repo_slug('/'.join([user, name]))
                         return
-                    elif url.startswith('git@'):
-                        if url.endswith('.git'):
-                            url = url[:-4]
+                    elif url.find('@') > -1 and url.find(':') > -1:
```
Comment (guyzmo) [inline git_repo/repo.py]: so you know, this is not very pythonic, the pythonic way is `if '@' in url` 😉

Comment (kounoike) [inline git_repo/repo.py]: Oh, thanks for comments!

**Pull request closed without merging**